referenceframe/frame_system.go

```diff
@@ -168,13 +168,8 @@ func (sfs *simpleFrameSystem) Parent(frame Frame) (Frame, error) {
 
 // frameExists is a helper function to see if a frame with a given name already exists in the system.
 func (sfs *simpleFrameSystem) frameExists(name string) bool {
-	if name == World {
-		return true
-	}
-	if _, ok := sfs.frames[name]; ok {
-		return true
-	}
-	return false
+	_, ok := sfs.frames[name]
+	return ok || name == World
```

Comment:
I didn't actually mean to commit this but now that I did I would like to keep it.  Should be a 1:1 change with the initial function, done with fewer instructions.

Reply:
Would like to note: this is not quite a 1:1 change.

Previously, if either of `sfs` or `sfs.frames` were nil, if queried with "World" we would not panic; now we will.

Since this is a private function I don't think this actually matters at all, as it can only be called in a controlled manner.

Reply:
Haha yes this is technically true.  I'm not worried about this because its not possible outside this package.  

Catching the world case first is slower because 90% of the time in practice we never actually call `fs.Frame("world")` we would call `fs.World()` directly.  Noticed this when debugging the race condition earlier in the week

Reply:
Actually- branch prediction means that the original version would be *ever so slightly* faster since the "if world" case gets assumed-false by the branch predictor and run in parallel, while now we're forcing a map lookup always even if the query *is* world. On a modern (post 2008) CPU simple if-statements like this are not just cheap but literally free.

That said as you note it doesn't matter because this function should almost never be called with world, and even if it does this is extremely not perf-sensitive.